src/fights.ts

```diff
@@ -733,6 +736,44 @@ class FreeFight {
   }
 }
 
+class FreeRunFight {
+  available: () => number | boolean;
+  run: (runSource: FreeRun) => void;
+  options: FreeFightOptions;
+
+  constructor(
+    available: () => number | boolean,
+    run: (runSource: FreeRun) => void,
+    options: FreeFightOptions = {}
+  ) {
+    this.available = available;
+    this.run = run;
+    this.options = options;
+  }
+
+  runAll() {
+    if (!this.available()) return;
+    // FIXME: make a better decision here.
+    if ((this.options.cost ? this.options.cost() : 0) > 2000) return;
+    while (this.available()) {
+      const runSource = findRun(this.options.familiar ? false : true);
+      if (!runSource) break;
+      useFamiliar(
+        this.options.familiar ? this.options.familiar() ?? freeFightFamiliar() : freeFightFamiliar()
+      );
+      if (runSource.prepare) runSource.prepare();
+      freeFightMood().execute();
+      freeFightOutfit([
+        ...(this.options.requirements ? this.options.requirements() : []),
+        ...(runSource.requirement ? [runSource.requirement] : []),
+      ]);
+      safeRestore();
+      // Abort and force the run function to use the correct macro
```

Comment:
this feels like a suboptimal solution

Reply:
upon review, nevermind, although i'm not quite sure why we want the withMacro wrapper here

Reply:
The withMacro wrapper is taken directly from the other FreeFight class, I needed to pass in the free run macro so entries that do other combat logic first (ex extract jelly) can do that and run away afterwards. I can make the default just use the free run but I was thinking if I made it abort that would be an explicit halt for someone making a new entry instead of wasting a run if hey needed to implement custom combat logic.